update docker settings, remove pipenv, update django logging config

mjhea0 opened the pull request:
This PR:

1. Finalizes the Docker config for development. This will help speed up the development process since you can spin up Django, Webpack, Postgres, and Elasticsearch with a single command. It will also help move the project away from being dependent on buildpacks for deployment since everything can be run in a container.
1. Replaces Pipenv with a requirements file and virtualenv. Pipenv's dependency resolution is slow on resource constrained environments - like Docker.
1. The Django logging config has been updated as well to work correctly with stdout/stderr and Docker.

jacobwegner commented:
@mjhea0: I made a couple of tweaks to `README.md` that I found as I went and tried to spin things up from a fresh checkout of the repo.

I found out that `node_modules` from `sv-webpack` was getting clobbered by the mount of the repo root, so I changed that to only mount `static/src`:

https://github.com/scaife-viewer/scaife-viewer/pull/365/commits/dd484c8fc7600931841abf49b642ddccffd5689c

Since `static/src` would only be modified within `sv-webpack`, I then created a named volume that could be shared across the containers (which would allow `scaife-viewer` to pick up `webpack-stats.json` changes from `sv-webpack`):

https://github.com/scaife-viewer/scaife-viewer/pull/365/commits/18b91de356f40511e7a0db28370faeb7bfb8b1cb

jacobwegner commented:
@mjhea0 I'm also debugging the GCR authentication issues.

The `EC_SERVICE_ACCOUNT` env var doesn't have anything to do with GCR (it is only used to talk to the Eldarion Cloud API during deployments), but I've added a new GCR-specific GCP service account:

https://github.com/scaife-viewer/scaife-viewer/pull/365/commits/02591843df5a9a1fe830b51ca635859006c36458

jacobwegner commented:
@mjhea0: Looks like the builds are failing, so I'll hand this back over to you:

https://circleci.com/gh/scaife-viewer/scaife-viewer/956

mjhea0 commented on scaife_viewer/wsgi.py:
```diff
@@ -17,8 +17,10 @@ def setup():
     # will inherit it. gunicorn --preload is required for this to work.
-    cts.TextInventory.load()
-    print("Loaded text inventory")
+    DEBUG = bool(int(os.environ.get("DEBUG", "0")))
```
I don't believe Django has loaded the settings at this point yet.

jacobwegner replied:
Sorry for the confusion; I left comments out of turn.

I had meant that the fallback value should match `settings.DEBUG`, and I'd prefer it be `1` to optimize for local dev.

In general, we've tried to set defaults in `settings.py` for local dev and then override as-needed in production.

jacobwegner replied:
I just tested and I think we can safely do an import of Django settings (because of `os.environ.setdefault("DJANGO_SETTINGS_MODULE", "scaife_viewer.settings")`).

`from . import cts` will load Django (because `cts.collections` contains `from django.conf import settings`).

mjhea0 commented on scaife_viewer/settings.py:
```diff
@@ -7,7 +8,7 @@
 BASE_DIR = PACKAGE_ROOT
 
-DEBUG = bool(int(os.environ.get("DEBUG", "1")))
+DEBUG = bool(int(os.environ.get("DEBUG", "0")))
```
Having a default of true for a production environment seems a bit dangerous.

jacobwegner replied:
I'd like to keep the default to `True` here and within `scaife_viewer.wsgi` in keeping with what we typically do for other projects.

Our "defaults" in settings.py are optimized for local dev and we then override with the proper env vars for production.

mjhea0 commented:
> @mjhea0 I'm also debugging the GCR authentication issues.
> 
> The `EC_SERVICE_ACCOUNT` env var doesn't have anything to do with GCR (it is only used to talk to the Eldarion Cloud API during deployments), but I've added a new GCR-specific GCP service account:
> 
> [0259184](https://github.com/scaife-viewer/scaife-viewer/commit/02591843df5a9a1fe830b51ca635859006c36458)

I was just guessing on the service account. Thought maybe it had access to GCR as well. Thanks for setting a new one up.

mjhea0 commented:
need to rebase!

jacobwegner commented on .circleci/config.yml:
```diff
@@ -1,39 +1,80 @@
+          command: |
+            apk add --no-cache 'py2-pip' docker
+            pip install 'docker-compose==1.23.2'
+      # - run:
```
See my overall review comment, but if we're not enabling these "steps", can we please remove them (and grab 'em from VCS history) rather than landing commented out code?

mjhea0 commented:
@jacobwegner:

1. Fixed the settings: `DEBUG = bool(int(os.environ.get("DEBUG", "1")))`
2. Added a description to the PR body
3. Removed Docker from the CI process
4. Added a new issue for adding Docker back to CI and troubleshooting the caching issue (https://github.com/scaife-viewer/scaife-viewer/issues/368)
